veracruz-mcu-client/vc.c

```diff
@@ -0,0 +1,577 @@
+/*
+ * Veracruz client aimed at microcontrollers, built in Zephyr OS
+ *
+ * ## Authors
+ *
+ * The Veracruz Development Team.
+ *
+ * ## Licensing and copyright notice
+ *
+ * See the `LICENSE.markdown` file in the Veracruz root directory for
+ * information on licensing and copyright.
+ *
+ */
+
+#include "vc.h"
+
+#include <random/rand32.h>
+#include <kernel.h>
+
+#include <mbedtls/ssl.h>
+#include <mbedtls/debug.h>
+
+#include "nanopb/pb.h"
+#include "nanopb/pb_encode.h"
+#include "nanopb/pb_decode.h"
+#include "transport_protocol.pb.h"
+
+#include "policy.h"
+#include "xxd.h"
+#include "base64.h"
+#include "http.h"
+#include "clap.h"
+
+
+//// Veracruz session handling ////
+static void mbedtls_debug(void *ctx, int level,
+        const char *file, int line,
+        const char *str) {
+    if (level <= 1) {
+        printf("%s", str);
+    }
+}
+
+static int vc_rawrng(void *p,
+        uint8_t *buf, size_t len) {
+    // TODO use cryptographically secure rng?
+    sys_rand_get(buf, len);
```

Comment:
Where is `sys_rand_get` defined? Is it a cryptographically secure RNG?

Reply:
Will look into this, `sys_rand_get` is NOT cryptographically secure, but I'm not sure if Zephyr provides a cryptographically secure rng vs leaving it as an exercise for the user

Reply:
Adopted `sys_csrand_get` was an easy drop in replacement. Well except the qemu_cortex_m3 build is now failing due to a linker error even with Zephyr's TEST_RANDOM_GENERATOR enabled... Not sure what that's about so just added a #ifdef dependent on TEST_RANDOM_GENERATOR. (Zephyr outputs a number of warnings at compile time if you enable this).